File: Fragment/Fragment_demo/app/src/main/java/com/example/fragment_demo/fragment/FragmentA.java
package com.example.fragment_demo.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.fragment_demo.R;

public class FragmentA extends Fragment {

    public FragmentA() {
        // Required empty public constructor
    }

    // thực hiện giao diện (View), trả view là giao diện file XML tương ứng Fragment
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_a, container, false);
    }
}

File: Fragment/Fragment_Api/app/src/main/java/com/example/fragment_api/Fragment_a.kt
package com.example.fragment_api

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.FragmentResultListener
import androidx.fragment.app.setFragmentResultListener
import kotlinx.android.synthetic.main.fragment_a.*


// nhan kêt quả
class Fragment_a : Fragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        childFragmentManager.setFragmentResultListener(
            REQUEST_KEY,
            this, FragmentResultListener{ requestKey, result ->
                val number = result.getInt(KEY_NUMBER)
                edt_address.text = getString (R.string.app_name, number)
            })
            Fragment_a.beginTransaction().add(R.id.fragment_a, Fragment_a::class.java, null).commit()
    }

    companion object {
        const val REQUEST_KEY = "parent-request-key"
        const val KEY_NUMBER = "key-number"
    }
}


